pop_tools/budget.py

```diff
@@ -0,0 +1,399 @@
+    """Compute contribution from vertical mixing."""
+    ds = _subset_slice_mask(ds, var_list=['DIA_IMPVF', 'KPP_SRC'], kmax=kmax, mask=mask)
+
+    # Only need to flip sign of DIA_IMPVF.
+    ds['DIA_IMPVF'] = _convert_to_tendency(ds['DIA_IMPVF'], grid, sign=-1, force_area=True)
+    ds['KPP_SRC'] = _convert_to_tendency(ds['KPP_SRC'], grid, kmax=kmax)
+
+    # Compute divergence of diapycnal mixing
+    diadiff = _compute_vertical_divergence(ds.DIA_IMPVF)
+
+    # Sum KPP and diapycnal mixing to create vmix term.
+    vmix = (ds.KPP_SRC + diadiff).rename('vmix')
+    vmix = _convert_units(vmix)
+    vmix.attrs['long_name'] = 'vertical mixing'
+    return vmix
+
+
+def _compute_SMS(ds, grid, mask, kmax=None):
+    """Compute SMS term from biology."""
+    ds = _subset_slice_mask(ds, var_list=['SMS'], kmax=kmax, mask=mask)
+    ds = _convert_to_tendency(ds, grid, kmax=kmax)
+    ds = _convert_units(ds)
+    ds = ds.rename({'SMS': 'sms'})
+    ds.attrs['long_name'] = 'source/sink'
+    return ds
+
+
+def _compute_surface_flux(ds, grid, mask):
+    """Computes virtual fluxes of tracer."""
+    ds = _subset_slice_mask(ds, var_list=['STF'], mask=mask)
+    ds = _convert_to_tendency(ds, grid)
+    ds = _convert_units(ds)
+    stf = ds.rename({'STF': 'stf'})
+    stf.attrs['long_name'] = 'surface tracer flux'
+    return stf
+
+
+def _compute_virtual_flux(ds, grid, mask):
+    """Computes virtual fluxes of tracer."""
+    ds = _subset_slice_mask(ds, var_list=['FvICE', 'FvPER'], mask=mask)
+    ds = _convert_to_tendency(ds, grid)
+    ds = _convert_units(ds)
+    vf = (ds.FvICE + ds.FvPER).rename('vf')
+    vf.attrs['long_name'] = 'virtual flux'
+    return vf
+
+
+def _process_input_dataset(ds):
+    """Checks that input dataset has appropriate variables, etc."""
+    mandatory_vars = ['UE', 'VN', 'WT', 'HDIFE', 'HDIFN', 'HDIFB', 'DIA_IMPVF', 'KPP_SRC']
+    if not set(mandatory_vars).issubset(ds):
+        missing_vars = [v for v in mandatory_vars if v not in ds]
+        error_msg = (
+            'Input dataset does not contain the mandatory variables for budget analysis. '
+            + f'`ds` is missing {missing_vars}'
+        )
+        raise IOError(error_msg)
+
+    # Drop coordinates for shift, roll, etc.
+    coord_vars = [c for c in ds.coords if c != 'time']
+    ds = ds.drop(coord_vars)
+    if 'z_w_top' in ds.dims:
+        ds = ds.drop('z_w_top')
+        ds = ds.rename({'z_w_top': 'z_t'})
+    if 'z_w_bot' in ds.dims:
+        ds = ds.drop('z_w_bot')
+        ds = ds.rename({'z_w_bot': 'z_t'})
+
+    # Force datetime
+    if not np.issubdtype(ds.time.dtype, np.datetime64):
+        raise TypeError('Please input a dataset with a time coordinate of type datetime.')
+
+    # Force chunking.
+    if not ds.chunks:
+        raise IOError('Please input a dataset with chunks (i.e., dask arrays).')
+    return ds
+
+
+def regional_tracer_budget(
+    ds, grid, mask=None, mask_int=None, budget_depth=None, sum_depth=True, sum_area=True
```

Comment:
Change to `compute_tracer_budget`. If `mask=None`, compute global. Drop `mask_int`. Should expect 3D masks where 1st dimension is int, second is nlat, third is nlon. (Per conversation with @mclong).